### PR TITLE
Expanded auth system

### PR DESCRIPTION
**Pull request by aaannz**
Contrary to the branch name this does not contain nonblocking openid. Instead this introduces plugin based authentication system together with plugins for OpenID, iChain and Fake auth (for development purposes).

### CODE REVIEW

File: lib/OpenQA/Auth/Fake.pm [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+sub auth_logout {
+    return;
+}
+
+sub auth_login {
+    my ($self) = @_;
+    my $headers = $self->req->headers;
+    my $username = 'FakeUser';
+    my $email = 'fake@user.org';
```
Comment (lnussel) [inline lib/OpenQA/Auth/Fake.pm]: How about using "Admin" or "Demo"? The reason I came up with fake auth was demoing openQA in front of "customers". I fear "FakeUser" looks a bit odd in such situation :)

**Comment by lnussel** [comment]
Are you still storing the full openid url for a user? That is needed in case the openid provider is switched.

**Comment by aaannz** [comment]
No, I don't store openid url anymore, just username.
I was thinking about provider switch, but I don't think that is properly supported anyway. When you switch provider and compare urls, you'll end up with limited account without admins and operators, since Users.pm:94 won't kick in.
But I agree this might end in dangerous situation when not done carefully. I plan to document that only in documentation though. For our purposes I wanted to make iChain and OpenID switchable without need to recreate users and permissions.

File: lib/OpenQA/Auth/Fake.pm (hunk 1 above)
Comment (aaannz) [inline lib/OpenQA/Auth/Fake.pm]: Will do

**Comment by aaannz** [comment]
I restored storing of openid identity urls in username column and removed any attempts to automatically match username to identity urls. In docu I thus plan to add note if admin wants to change auth method, she should prepare users db migration by herself.

**Comment by coolo** [comment]
I'd tend to merge this (especially as I need to rebase my branch once more for one more migration), but I think what we're missing is documentation on how to set this up.

**Comment by aaannz** [comment]
I'll add the proper docu today, but in few words: For OpenID auth there is no change needed. To activate different auth methods just change [auth] method option in openqa.ini.

**Comment by coolo** [comment]
but fake auth is the one we should default to in git IMO.

**Comment by nilxam** [comment]
I've a small concern, removed openid section from defaults, actually the code below will ignore your openid config in the openqa.ini
my $v = $cfg && $cfg->val($section, $k);

for example it will always httpsonly even I sets httpsonly=0 in openqa.ini...doesn't it?

**Comment by nilxam** [comment]
hmm..I've opened https://github.com/os-autoinst/openQA/pull/166
that is just what I did on my machine, not so sure is it good or bad for your authentication changes/plan, but we can discuss... :)

**Pull request merged**